src/com/facebook/buck/go/GoBinaryDescription.java

```diff
@@ -0,0 +1,111 @@
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.go;
+
+import com.facebook.buck.cxx.CxxPlatform;
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.model.ImmutableFlavor;
+import com.facebook.buck.rules.BuildRule;
+import com.facebook.buck.rules.BuildRuleParams;
+import com.facebook.buck.rules.BuildRuleResolver;
+import com.facebook.buck.rules.BuildRuleType;
+import com.facebook.buck.rules.Description;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.SourcePathResolver;
+import com.facebook.buck.rules.TargetGraph;
+import com.facebook.infer.annotation.SuppressFieldNotInitialized;
+import com.google.common.base.Optional;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSortedSet;
+
+import java.nio.file.Paths;
+import java.util.List;
+
+public class GoBinaryDescription implements Description<GoBinaryDescription.Arg> {
+
+  private static final BuildRuleType TYPE = BuildRuleType.of("go_binary");
+
+  private final GoBuckConfig goBuckConfig;
+
+  private final CxxPlatform cxxPlatform;
+
+  public GoBinaryDescription(GoBuckConfig goBuckConfig, CxxPlatform cxxPlatform) {
+    this.goBuckConfig = goBuckConfig;
+    this.cxxPlatform = cxxPlatform;
+  }
+
+  @Override
+  public BuildRuleType getBuildRuleType() {
+    return TYPE;
+  }
+
+  @Override
+  public Arg createUnpopulatedConstructorArg() {
+    return new Arg();
+  }
+
+  @Override
+  public <A extends Arg> BuildRule createBuildRule(
+      TargetGraph targetGraph,
+      BuildRuleParams params,
+      BuildRuleResolver resolver,
+      A args) {
+    SourcePathResolver pathResolver = new SourcePathResolver(resolver);
+
+    BuildTarget libraryTarget =
+        BuildTarget.builder(params.getBuildTarget())
+            .addFlavors(ImmutableFlavor.of("lib"))
+            .build();
+
+    GoLibrary lib = new GoLibrary(
+        params.copyWithBuildTarget(libraryTarget),
+        pathResolver,
+        Optional.of(Paths.get("main")),
+        ImmutableSet.copyOf(args.srcs),
+        ImmutableList.<String>builder()
+            .addAll(this.goBuckConfig.getCompilerFlags())
+            .addAll(args.compilerFlags.get())
+            .build(),
+        goBuckConfig.getGoCompiler().get()
+    );
+    resolver.addToIndex(lib);
+
+    return new GoBinary(
+        params.copyWithDeps(
+            Suppliers.ofInstance(ImmutableSortedSet.<BuildRule>of(lib)),
```

Comment:
This will end up removing the existing deps.

You need to add `lib` to the existing deps.